### PR TITLE
eth/tracers, internal/ethapi: remove unnecessary map pointer in state override

### PR DESCRIPTION
**Pull request by caseylove**
1. `map` is reference type which needn't use pointer
2. Since this PR only affects the `internal` package such won't break public API

### CODE REVIEW

File: internal/ethapi/api.go [hunk 1]
```diff
@@ -968,11 +968,11 @@ func (api *BlockChainAPI) GetBlockReceipts(ctx context.Context, blockNrOrHash rp
 // if stateDiff is set, all diff will be applied first and then execute the call
 // message.
 type OverrideAccount struct {
-	Nonce     *hexutil.Uint64              `json:"nonce"`
-	Code      *hexutil.Bytes               `json:"code"`
-	Balance   **hexutil.Big                `json:"balance"`
-	State     *map[common.Hash]common.Hash `json:"state"`
-	StateDiff *map[common.Hash]common.Hash `json:"stateDiff"`
```
Comment (karalabe) [inline internal/ethapi/api.go:975]: I think the reason was that we wanted to have everything optional, but missing or `null` json *can* decode into nil map, so should be fine. I think balance can also lose a pointer with the same logic.

Comment (caseylove) [reply internal/ethapi/api.go:975]: Thx sir. `Balance` field already remove a pointer now.

**Comment by caseylove** [comment]
@rjl493456442 @lilasxie Hi sir, sorry to bother you two, It seems that #29901 doestn't fix the issue of `TestDisable`.

See failed job: https://ci.appveyor.com/project/ethereum/go-ethereum/builds/50137333/job/s6oluctjxj85ejqi

Failed logs
```shell
--- FAIL: TestDisable (0.21s)
    database_test.go:494: Invalid activation should be rejected
FAIL
FAIL	github.com/ethereum/go-ethereum/triedb/pathdb	3.557s
FAIL
util.go:47: exit status 1
exit status 1
Command exited with code 1
```

**Comment by caseylove** [comment]
> Hi sir, sorry to bother you two, It seems that #29901 doestn't fix the issue of TestDisable.

> See failed job of latest commit of this PR: https://ci.appveyor.com/project/ethereum/go-ethereum/builds/50137333/job/s6oluctjxj85ejqi


@karalabe Hi sir, anything else need I update? I think the ci issue above is not caused by this pr.

**Review by rjl493456442 — APPROVED** [review]

**Pull request merged**